test/cpp/end2end/BUILD

```diff
@@ -943,8 +943,8 @@ grpc_cc_test(
     srcs = ["grpc_authz_end2end_test.cc"],
     data = [
         "//src/core/tsi/test_creds:ca.pem",
-        "//src/core/tsi/test_creds:client.key",
-        "//src/core/tsi/test_creds:client.pem",
+        "//src/core/tsi/test_creds:client3.key",
```

Comment:
I see we've totally replaced the `client1` here, just want to verify that we aren't losing any coverage because of this, and `client3` is strictly a superset of what `client1` was testing?
Otherwise we can keep the old test with the old certs as well

Reply:
`client.pem` is simply a cert signed by the same CA without any x509 extensions. The new cert is a strict superset with the URI SAN added. The existing test doesn't check anything inside the cert other than verifying it. We don't need the old test.